JLio.Commands/Advanced/Compare.cs

```diff
@@ -0,0 +1,482 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using JLio.Commands.Advanced.Models;
+using JLio.Commands.Advanced.Settings;
+using JLio.Commands.Builders;
+using JLio.Commands.Logic;
+using JLio.Core;
+using JLio.Core.Contracts;
+using JLio.Core.Models;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace JLio.Commands.Advanced
+{
+    public class Compare : CommandBase
+    {
+        private IExecutionOptions options;
+
+        [JsonProperty("firstPath")]
+        public string FirstPath { get; set; }
+
+        [JsonProperty("resultPath")]
+        public string ResultPath { get; set; }
+
+        [JsonProperty("secondPath")]
+        public string SecondPath { get; set; }
+
+        [JsonProperty("settings")]
+        public CompareSettings Settings { get; set; }
+
+        public override ValidationResult ValidateCommandInstance()
+        {
+            var result = new ValidationResult {IsValid = true};
+            if (string.IsNullOrWhiteSpace(FirstPath))
+            {
+                result.ValidationMessages.Add($"FirstPath property for {CommandName} command is missing");
+                result.IsValid = false;
```

Comment:
change ValidationResult implementation so that the boolean IsValid is automatically looking if there is any ValidationMessage,  that way you don't have to explicitly set it to false every time.  In case you forget to do this, it's going to be a mismatch, and code could wrongly rely on the boolean to make decisions